### PR TITLE
Bug report graph line leads

### PR DESCRIPTION
**Pull request by tomekkowalczyk**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                |  features
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:
If we create a report based on contacts from the segment:
1. Data source: Contacts
2. Filters equals Segment
image

and we add "New contacts added (Line graph)" we have data not compatible with the table

image

The query to the database looks like this  
```
SELECT DATE_FORMAT(l.date_added, '%Y-%m') AS date, COUNT(*) AS count FROM leads l 
INNER JOIN lead_lists_leads s 
ON s.lead_id = l.id 
AND s.manually_removed = 0 
WHERE (s.date_added IS NULL OR (s.date_added BETWEEN :dateFrom AND :dateTo)) 
AND (s.leadlist_id = :i4csleadlistid)
GROUP BY DATE_FORMAT(l.date_added, '%Y-%m') 
ORDER BY DATE_FORMAT(l.date_added, '%Y-%m') 
ASC LIMIT 4
```

In my opinion it should look like this
```
SELECT DATE_FORMAT(s.date_added, '%Y-%m') AS date, COUNT(*) AS count FROM leads l 
INNER JOIN lead_lists_leads s 
ON s.lead_id = l.id 
AND s.manually_removed = 0 
WHERE (s.date_added IS NULL OR (s.date_added BETWEEN :dateFrom AND :dateTo)) 
AND (s.leadlist_id = :i4csleadlistid) 
GROUP BY DATE_FORMAT(s.date_added, '%Y-%m') 
ORDER BY DATE_FORMAT(s.date_added, '%Y-%m') 
ASC LIMIT 4

```
Correct line graph
image

### CODE REVIEW

**Review by kuzmany — CHANGES_REQUESTED** [review]
Doesn't work If don't use segment filter:


` 500 Internal Server Error - An exception occurred while executing 'SELECT DATE_FORMAT(s.date_added, '%Y-%m-%d') AS date, COUNT(*) AS count FROM 9547_leads l WHERE l.date_added IS NULL OR (l.date_added BETWEEN ? AND ?) GROUP BY DATE_FORMAT(s.date_added, '%Y-%m-%d') ORDER BY DATE_FORMAT(s.date_added, '%Y-%m-%d') ASC LIMIT 32' with params ["2020-12-01 00:00:00", "2021-01-01 20:13:43"]: SQLSTATE[42S22]: Column not found: 1054 Unknown column 's.date_added' in 'field list' `

You can test it here: https://mautibox.com/9547/s/reports/view/11

**Review by kuzmany — CHANGES_REQUESTED** [review]
Doesn't work If don't use segment filter:


` 500 Internal Server Error - An exception occurred while executing 'SELECT DATE_FORMAT(s.date_added, '%Y-%m-%d') AS date, COUNT(*) AS count FROM 9547_leads l WHERE l.date_added IS NULL OR (l.date_added BETWEEN ? AND ?) GROUP BY DATE_FORMAT(s.date_added, '%Y-%m-%d') ORDER BY DATE_FORMAT(s.date_added, '%Y-%m-%d') ASC LIMIT 32' with params ["2020-12-01 00:00:00", "2021-01-01 20:13:43"]: SQLSTATE[42S22]: Column not found: 1054 Unknown column 's.date_added' in 'field list' `

You can test it here: https://mautibox.com/9547/s/reports/view/11

File: app/bundles/LeadBundle/EventListener/ReportSubscriber.php [hunk 1]
```diff
@@ -481,7 +481,7 @@ public function onReportGraphGenerate(ReportGraphEvent $event)
 
                 case 'mautic.lead.graph.line.leads':
                     $chart = new LineChart(null, $options['dateFrom'], $options['dateTo']);
-                    $chartQuery->modifyTimeDataQuery($queryBuilder, 'date_added', 'l');
+                    $chartQuery->modifyTimeDataQuery($queryBuilder, 'date_added', 's');
```
Comment (kuzmany) [inline app/bundles/LeadBundle/EventListener/ReportSubscriber.php]: You should add tablePrefix s just If query contain lead_lists_leads table

**Review by npracht — APPROVED** [review]
Well done !
Capture d’écran 2021-01-20 à 09 43 31

**Comment by RCheesley** [comment]
I think that this has already been fixed - or I am not understanding the instructions.

Using 3.2.5-Release Candidate I see a correct line chart that corresponds with the number of contacts added:

screenshot-mautic32 ddev site-2021 01 22-17_22_02

@npracht @tomekkowalczyk could you please confirm?

**Review by kuzmany — APPROVED** [review]
Works for me. 
Code looks good 👍

**Comment by kuzmany** [comment]
@RCheesley this issue is related to report's contacts line graph

image

**Pull request merged**